src/Sylius/Bundle/ResourceBundle/Controller/RequestConfiguration.php

```diff
@@ -326,66 +322,128 @@ public function getRequestParameter($parameter, $defaults = array())
         );
     }
 
-    public function getRepositoryMethod($default)
+    /**
+     * @return string|null
+     */
+    public function getRepositoryMethod()
     {
-        $repository = $this->parameters->get('repository', array('method' => $default));
+        if (!$this->parameters->has('repository')) {
+            return null;
+        }
+
+        $repository = $this->parameters->get('repository');
 
         return is_array($repository) ? $repository['method'] : $repository;
     }
 
-    public function getRepositoryArguments(array $default = array())
+    /**
+     * @return array
+     */
+    public function getRepositoryArguments()
     {
-        $repository = $this->parameters->get('repository', array());
+        if (!$this->parameters->has('repository')) {
+            return array();
+        }
+
+        $repository = $this->parameters->get('repository');
 
-        return isset($repository['arguments']) ? $repository['arguments'] : $default;
+        if (!isset($repository['arguments'])) {
+            return array();
+        }
+
+        return is_array($repository['arguments']) ? $repository['arguments'] : array($repository['arguments']);
     }
 
-    public function getFactoryMethod($default)
+    /**
+     * @return string|null
+     */
+    public function getFactoryMethod()
     {
-        $factory = $this->parameters->get('factory', array('method' => $default));
+        if (!$this->parameters->has('factory')) {
+            return null;
+        }
+
+        $factory = $this->parameters->get('factory');
 
         return is_array($factory) ? $factory['method'] : $factory;
```

Comment:
No, but I will leave that for proper configuration validation PR sometime soon. :)